visual_behavior/visualization/ophys/summary_figures.py

```diff
@@ -1208,13 +1706,343 @@ def plot_cell_summary_figure(analysis, cell_index, save=False, show=False, cache
     ax = placeAxesOnGrid(fig, dim=(1, 2), xspan=(0.0, 0.5), yspan=(.78, .99), wspace=0.25, sharex=True, sharey=True)
     ax = plot_mean_response_pref_stim_metrics(analysis, cell_index, ax=ax, save=False, use_events=use_events)
 
+    # # ax = placeAxesOnGrid(fig, dim=(1, 1), xspan=(.83, 1), yspan=(.78, 1))
+    # ax = placeAxesOnGrid(fig, dim=(1, 1), xspan=(.7, .99), yspan=(0.05, .16))
+    # table_data = format_table_data(dataset)
+    # xtable = ax.table(cellText=table_data.values, cellLoc='left', rowLoc='left', loc='center', fontsize=12)
+    # xtable.scale(1, 3)
+    # ax.axis('off');
+
+    fig.canvas.draw()
     fig.tight_layout()
-    plt.gcf().subplots_adjust(bottom=0.05)
+    # plt.gcf().subplots_adjust(bottom=0.05)
     if save:
         save_figure(fig, figsize, analysis.dataset.analysis_dir, 'cell_summary_plots',
                     str(cell_index) + '_' + str(cell_specimen_id) + suffix)
         if cache_dir:
             save_figure(fig, figsize, cache_dir, 'cell_summary_plots',
                         dataset.analysis_folder + '_' + str(cell_specimen_id) + '_' + str(cell_index) + suffix)
-        if not show:
-            plt.close()
+    if not show:
+        plt.close()
+
+
+def colormap():
+    colormap = {
+        'engaged': 'olivedrab',
+        'disengaged': 'firebrick'
+    }
+    return colormap
+
+
+def get_title(ophys_experiment_id, cell_specimen_id):
+    '''
+    generate a standardized figure title containing identifying information
+    '''
+    experiments_table = loading.get_filtered_ophys_experiment_table().reset_index()
+
+    row = experiments_table.query('ophys_experiment_id == @ophys_experiment_id').iloc[0].to_dict()
+    title = '{}__specimen_id={}__exp_id={}__{}__{}__depth={}__cell_id={}'.format(
+        row['cre_line'],
+        row['specimen_id'],
+        row['ophys_experiment_id'],
+        row['session_type'],
+        row['targeted_structure'],
+        row['imaging_depth'],
+        cell_specimen_id,
+    )
+    return title
+
+
+def merge_in_extended_stimulus_presentations(analysis):
```

Comment:
@dougollerenshaw the ResponseAnalysis object will merge extended_stimulus_presentations with the response dataframes automatically if you set the use_extended_stimulus_presentations flag to True in the call to create ResponseAnalysis. Is there a reason to have this additional function to do this task? we might want to consider removing this and using the built in functionality, unless that doesnt meet the need you had in mind when you wrote this function.

Reply:
Oh, nice. I totally overlooked that functionality. I'll modify to take advantage.

Reply:
@matchings it looks like the extended stimulus presentations aren't being merged into the `trials_response_df`

Here's code to load an analysis object
```
from visual_behavior.ophys.response_analysis.response_analysis import ResponseAnalysis
import visual_behavior.data_access.loading as loading

experiments_table = loading.get_filtered_ophys_experiment_table()
experiment_ids = experiments_table.query('model_outputs_available == True').index.values

experiment_id = experiment_ids[0]
dataset = loading.get_ophys_dataset(experiment_id)

analysis = ResponseAnalysis(dataset, overwrite_analysis_files=True, use_extended_stimulus_presentations=True) 
```

I can confirm that the extended stim presentations has been merged with the `stimulus_response_df` as follows:

`'engagement_state' in analysis.stimulus_response_df.columns`

Which returns` True`

But when I call:

`'engagement_state' in analysis.trials_response_df.columns`

I get `False`

I'll dig in and try to figure out how you're making the merge happen so we can also merge it into `trials_response_df`. But if you have any quick ideas (or if I'm doing something wrong), let me know.

Reply:
This behavior makes sense, because the `trials_response_df `gets merged with the `trials `table not `stimulus_presentations_table `or `extended_stimulus_presentations_table`. We would need to add 'engagement_state' to the `trials `table as well. So I guess my original statement that all the functionality you had there was already covered by `ResponseAnalysis `was not entirely accurate

Reply:
I just pushed a commit that solves this. I'm merging `extended_stimulus_presentations` into the `trials_df` on the `change_frame` and `start_time` columns. So the extended stimulus presentations attributes for the change flash will be assigned to the entire trial (e.g., if the enagement_state is 'engaged' on the change flash, the enagagement state will be 'engaged for that entire trial, even if the engagment state was different for any of the other flashes in the trial)

Reply:
Im so entirely sure about this. There are a ton of things in `extended_stimulus_presentations`, some of which we may not want in the trials table. Here are all the extra columns that get added: 

'duration', 'end_frame', 'image_index', 'image_name', 'omitted',
       'start_frame', 'start_time_duplicate', 'stop_time_duplicate', 'change',
       'epoch_duplicate', 'mean_running_speed', 'licks', 'rewards',
       'time_from_last_lick', 'time_from_last_reward', 'time_from_last_change',
       'time_from_last_omission', 'flash_after_omitted', 'flash_after_change',
       'licked', 'lick_rate', 'rewarded', 'reward_rate', 'running_rate',
       'bout_rate', 'hit_bout', 'hit_fraction', 'high_lick', 'high_reward',
       'flash_metrics_epochs', 'flash_metrics_labels', 'in_bout', 'bias',
       'omissions1', 'task0', 'timing1D', 'engagement_state',
       'lick_on_next_flash', 'lick_rate_next_flash'

For example, I dont think 'omitted' makes sense to include, or 'flash_after_change', or 'image_name', given that 'change_image_name is already there and now its confusing/duplicated info. To me, it makes more sense to only add the specific columns that we want. 

I also think its an issue that people might infer that some of the column metrics are computed across the whole trial or the +/-5 second window corresponding to the trace, when they are actually just for the single change flash. Such as `mean_running_speed `or `lick_rate`